docs/DeveloperGuide.adoc

```diff
@@ -279,87 +504,232 @@ Refer to the guide <<DevOps#, here>>.
+2.  Calgo creates a user-friendly text file `FoodRecord.txt` containing all `Food` item details in the `exports` folder.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+[discrete]
+=== Use case: `list` all current `Food` entries
+
+*MSS*
+
+1.  User requests to `list` all current `FoodRecord` entries.
+2.  `Calgo` shows a list of all `Food` items in the current `FoodRecord`.
 +
+Use case ends.
+
+*Extensions*
+
 [none]
-** 3a1. AddressBook shows an error message.
+*The `FoodRecord` is empty.* +
+`Calgo` shows a message indicating that the `FoodRecord` is currently empty. +
+Use case ends.
+
+[discrete]
+=== Use case: `update` current `FoodRecord` with a new `Food` item
+
+*MSS*
+
+1.  User requests to add a new `Food` item in the `FoodRecord`.
+2.  `Calgo` creates and saves a new `Food` item in the `FoodRecord` with nutritional information specified by user.
++
+Use case ends.
+
+[discrete]
+=== Use case: `update` an existing `Food` item in current `FoodRecord`
+
+*MSS*
+
+1.  User requests to edit an existing `Food` item in the `FoodRecord`.
+2.  `Calgo` replaces the existing `Food` item's nutritional values with the new information.
++
+Use case ends.
+
+[discrete]
+=== Use case: `delete` an existing `Food` item in current `FoodRecord`
+
+*MSS*
+
+1.  User requests to delete an existing `Food` item from the `FoodRecord`
+2.  `Calgo` deletes the existing `Food` item in the `FoodRecord`.
++
+Use case ends.
+
+[discrete]
+=== Use case: set a dietary `goal`
+
+*MSS*
+
+1.  User uses `goal` command to set a dietary `DailyGoal` for the daily number of `Calorie` s to be consumed.
+2.  `Calgo` stores this `DailyGoal` in user preferences and analyses `Food` consumption with respect to this `goal`.
++
+Use case ends.
+
+
+[discrete]
+=== Use case: generate a `report` on a specific day.
+
+*MSS*
+
+1.  User requests to obtain a `report` on his or her `Food` consumption patterns on a particular day.
+2.  Calgo analyses the `Food` consumed on that day and generates a document with actionable insights for the user.
 +
-Use case resumes at step 2.
+Use case ends.
 
-_{More to be added}_
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 Food items without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-_{More to be added}_
+.  Calgo should work on both 32-bit and 64-bit environments.
+.  The product expects users to initially find out about Food items and their respective nutritional values for creating Food item entries for the first time.
 
 [appendix]
 == Glossary
 
+[[food]] Food::
```

Comment:
Is it necessary to explain what a food is in the glossary? It might be better to explain what the Food class encompasses in one of the above sections.